File: src/main/java/com/hyak4j/Hello.java
package com.hyak4j;

public class Hello {
    public static void main(String[] args) {
        String s = "abcde";
        System.out.println(s.charAt(3));
        System.out.println(s.substring(1, 4));
        Person p = new Person("Henry", 66.5f, 1.7f);
        p.hello();
        System.out.println(p.bmi());
        int score = 88;
        System.out.println(score > 80 && score < 90);
        char c ='A'; // 65
        System.out.println(c > 65);
        /*int age = 19;
        Integer age2 = 19;
        char c = 'A';
        byte b = 120;
        float weight = 66.5f;
        double height = 1.7;
        boolean adult = true;
        boolean enroll = false;
        String name = "Henry";*/
    }
}

File: src/main/kotlin/com/hyak4j/kotlin/Hello.kt
package com.hyak4j.kotlin

fun main() {
//    println("Hello Kotlin!")
//    Person().hello()
    var s:String? = "abcde"
    s = null
    println(s?.length)
    println(s?.get(3))
    println(s?.substring(3))

    val h = Person(66.5f, 1.7f)
    h.hello()
    println(h.bmi())
    val score = 88;
    println(score > 60)

    val c : Char = 'A'
    println(c.code > 60)
/*    val age : Int = 19;
    var weight = 66.5f;
    var name : String;
    name = "Henry"*/
}

class Person(var weight: Float, var height: Float, var name: String = ""){ // 不重要的參數擺後面
    init {
        println("test $weight")
    }
    // 次要建構子
//    constructor(name: String, weight: Float, height: Float) : this(weight, height)

    fun bmi(): Float {
        val bmi = weight / (height * height)
        return bmi
    }
    fun hello(){
        println("Hello Kotlin!!")
    }
}
